bin/ruby-build

```diff
@@ -244,24 +244,33 @@ compute_md5() {
   fi
 }
 
+declare -a HAS_CHECKSUM_SUPPORT
+has_checksum_support() {
+  local checksum_command="$1"
+  return ${HAS_CHECKSUM_SUPPORT[$checksum_command]:=$(echo test | "$checksum_command" >/dev/null; echo $?)}
```

Comment:
Bash 3 doesn't support associative arrays, so I can't imagine `HAS_CHECKSUM_SUPPORT[$checksum_command]` working. What you could do as an alternative to an array is create variable names dynamically:

```sh
has_checksum_var="HAS_CHECKSUM_SUPPORT_${checksum_command}"
if [ -n "${!has_checksum_var+defined}" ]; then
  return "${!has_checksum_var}"
else
  local supported=0
  echo test | "$checksum_command" >/dev/null || supported=$?
  eval "${has_checksum_var}=${supported}"
fi
```